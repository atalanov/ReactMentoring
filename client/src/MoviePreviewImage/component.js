import React from 'react';
import { Link } from 'react-router-dom';

export default props => (
  <div className="col">
    <Link to={`/movies/${props.id}`}><img src={props.src} alt={props.title} width="100%" /></Link>
  </div>
);

