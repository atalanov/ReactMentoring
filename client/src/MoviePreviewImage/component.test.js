import React from 'react';
import { shallow, render } from 'enzyme';
import MoviePreviewImage from './component';

describe('MoviePreviewImage tests', () => {
    test('Component MoviePreviewImage renders correctly', () => {
        const component = shallow(<MoviePreviewImage src="/testurl" title="test title" />);
        expect(component).toMatchSnapshot();
    });
});
