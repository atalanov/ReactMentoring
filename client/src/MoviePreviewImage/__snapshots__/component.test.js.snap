// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePreviewImage tests Component MoviePreviewImage renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    src="/testurl"
    title="test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="/movies/undefined"
      >
        <img
          alt="test title"
          src="/testurl"
          width="100%"
        />
      </Link>,
      "className": "col",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <img
          alt="test title"
          src="/testurl"
          width="100%"
        />,
        "replace": false,
        "to": "/movies/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "test title",
          "src": "/testurl",
          "width": "100%",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="/movies/undefined"
        >
          <img
            alt="test title"
            src="/testurl"
            width="100%"
          />
        </Link>,
        "className": "col",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            alt="test title"
            src="/testurl"
            width="100%"
          />,
          "replace": false,
          "to": "/movies/undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "test title",
            "src": "/testurl",
            "width": "100%",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
